### ISSUE TITLE
RTL - tests timing out

### ISSUE THREAD

**Issue by fllprbt** [NickColley/jest-axe#119]
Hi @nickcolley!

first of all thanks for your contribution in a11y tooling, it definitely helps making apps more inclusive. Regarding my issue, some of my tests are timing out after having jest-aXe integrated in my React Testing Library suite. I do have a quite recent version of RTL so **cleanup** won't help.

It is interesting to point that tests would pass:

- `--runInBand` flag was used

Regarding the context of tests, all of them are "wrapper" components (like forms) and not bare elements, and they don't really make any extra assertions. They are just rendered and then passed to jest-aXe. It should be noted that prior to aXe's integration, their `it renders correctly` case would never have any issues. AFAIK, at the time of the timeout `<body />` is empty.

Have already tried fiddling with the `testTimeout` and `--max-old-space-size` to no avail.

Any ideas of what might be the culprit or steps to identify potential problems are more than welcome.

Thanks in advance!

**Comment by dkershner6** [comment]
We ran into this as well, and in our case it was because we were using `jest.useFakeTimers()`

Once we moved the jest-axe test to a separate file, it worked.

**Comment by fllprbt** [comment]
@dkershner6 thanks for the help but I have this issue on files which are not utilising useFakeTimers.

Splitted for the sake of testing but it's back... runInBand does help but not always...

**Comment by nickcolley** [comment]
I've seen this sort of thing happen when a different part of the application had a memory leak or didnt exit nicely, did you manage to make a reduced case with *just* jest-axe to confirm it's this library?

We could then use that reduced test case to come up with a fix.

**Comment by ortigozamatias** [comment]
I'm using `jest.useFakeTimers()` and got this error. However, since such function cannot be used in CodeSandbox, at least for now, I guess that providing a reproduction is quite difficult.

**Comment by dbasilio** [comment]
I created a separate issue for the fakeTimers issue https://github.com/nickcolley/jest-axe/issues/156 and included a workaround for it, but it seems unrelated to the issue @fllprbt is talking about if `--runInBand` fixes it

**Comment by Zach-Jaensch** [comment]
Any progress on this? We don't use `jest.useFakeTimers()` but we are still getting timeout issues. They occur more often in our CI/CD then locally, suspecting hardware performance is impacting it. 

Seems to have the most issues with our Form components. 

If it helps, here is a list of the related packages.

    "react": "17.0.2",
    "jest": "^26.6.3",
    "jest-axe": "^4.1.0",
    "formik": "^2.2.9",
    "@material-ui/core": "^4.11.4",
    "@testing-library/dom": "^7.31.2",
    "@testing-library/jest-dom": "^5.12.0",
    "@testing-library/react": "^11.2.7",
    "@testing-library/react-hooks": "^6.0.0",
    "@testing-library/user-event": "^13.1.9",

**Comment by fllprbt** [comment]
Hi! I have not conducted any profiling so take my current update with a grain of salt. We stopped having the issue **but** we did bump our timeout times and upgraded our unit tests' ecosystem heavily. AFAIK though, the drive for the timeout bumps was completely different so I assume that the issue was resolved beforehand already.

A bit out of context but we used to have a jest-aXe internal error when asserting the a11y of a specific component with a deep tree. I was regularly checking and today the jest-axe went through it without crushing! So I will post our setup in case it helps:

```
react@16.14.0
react-dom@16.14.0
jest@26.6.0
jest-environment-jsdom@26.6.2
jest-axe@5.0.1
ts-jest@26.5.6
@testing-library/jest-dom@5.15.1
@testing-library/react@12.1.2
@testing-library/user-event@13.5.0
```
